src/cdk/collections/selection-model.ts

```diff
@@ -238,6 +233,21 @@ export class SelectionModel<T> {
   private _hasQueuedChanges() {
     return !!(this._deselectedToEmit.length || this._selectedToEmit.length);
   }
+
+  /** Returns a value that is comparable to inputValue by applying compareWith function, returns the same inputValue otherwise. */
```

Comment:
I end up creating the `_getConcreteValue` that returns  any comparable selected value, or returns the value passed as a parameter.
`_getConcreteValue` is called before selection / deselection.